### PR TITLE
Dexsearch: Fix types, and randmove

### PR DESCRIPTION
**Pull request by CheeseMuffin**
Fix `/randmove` and, not allowing `/ds darktypeasd`

### CODE REVIEW

File: chat-plugins/datasearch.js [hunk 1]
```diff
@@ -434,7 +434,7 @@ function runDexsearch(target, cmd, canAll, message) {
 
 			let typeIndex = target.indexOf('type');
 			if (typeIndex === -1) typeIndex = target.length;
-			if (typeIndex !== target.length || toId(target) in allTypes) {
+			if ((typeIndex !== target.length && target.endsWith('type')) || toId(target) in allTypes) {
```
Comment (Zarel) [inline chat-plugins/datasearch.js]: This seems very unnecessarily convoluted.

File: chat-plugins/datasearch.js [hunk 2]
```diff
@@ -434,7 +434,7 @@ function runDexsearch(target, cmd, canAll, message) {
 
 			let typeIndex = target.indexOf('type');
```
Comment (Zarel) [inline chat-plugins/datasearch.js]: Why isn't this line replaced with a simple detection of `target.endsWith('type')`?

**Comment by CheeseMuffin** [comment]
Do these checks look better?

**Comment by Zarel** [comment]
I don't really know what this is doing anymore, but in my experience your dexsearch changes haven't really caused problems, so I guess I'm merging.

**Pull request merged**